project-koku/koku
# Adding OCI provider type

Fixes [COST-2393](https://issues.redhat.com/browse/COST-2393)

Changes proposed in this PR:
* Adding new provider type OCI

Testing Instructions:
1. Start koku
2. Post the following:
   ```
   {
    "name": "TestSource OCI",
    "source_type": "OCI-local",
    "authentication": {
        "credentials": {
            "tenant": "test-oci-tenant"
        }}}

3. Check source is created


NOTES!!
This is _ready_ but we can only add my dev account as a real source right now. This is due to the permissions issues on OCI side that I am still working on. (Currently blocked by OCI account upgrade) That being said there is no real reason to not merge this one as is. If you want to test adding a real source locally reach out to me and I can give you the correct authentication. You will also require the auth for koku to access the OCI service account.

The other thing left is adding these creds to the clowdapp etc. Not sure if we want to do that now or in a follow up PR later.

I also ran the smoke tests locally since the pr_check is having its usual deploy issues. Everything passed locally for me using `ENV_FOR_DYNACONF=local iqe tests plugin cost_management -k test_api -m cost_required`
